### ISSUE TITLE
Support SQLite

### ISSUE THREAD

**Issue by n1trux** [jrie/sqstorage#32]
I'd really like to run this on devices without a "fat" DB daemon, which is where SQLite would be beneficial.

**Comment by wernersbacher** [comment]
+1, it would make it really quick and easy.

**Comment by jrie** [comment]
I agree, it might be nice to have SQLite support - but at the moment, there are several database changes going on and possible coming and adding another database to work with, might be slowing down the development.

When things are more settled this is definitely something to look into.

**Comment by n1trux** [comment]
> at the moment, there are several database changes going on and possible coming and adding another database to work with, might be slowing down the development.

Or you could use an abstraction layer which works with both.

I found [Medoo](https://github.com/catfan/Medoo) on the [lightweight-php list](https://github.com/jm42/lightweight-php#database) which "supports all SQL databases, including MySQL, MSSQL, SQLite, MariaDB, PostgreSQL, Sybase, Oracle and more".

Changes to the database scheme could be implemented by properly versioning the database (related: #34) and creating migrations. These could easily be written with Medoo if there's a mismatch between installed DB version and upgraded DB version.

**Comment by schnoog** [comment]
Would prefer to wait for https://github.com/SergeyTsalkov/meekrodb/issues/66 to keep the current wrapper syntax alive.

**Comment by jrie** [comment]
So, very long time ago this issue was raised. I will see if this can be done without breaking any logic for MariaDB.
Because I just checked and `DB::$dsn = 'sqlite:';` is a thing.

Lets see. I keep you updated here. 👍

**Comment by jrie** [comment]
Just for sake of completeness - the database example configuration is inside "dba-example.php".
The exact path to the "prepared" database in `/support/sqlite_db.sq3` has to be defined/adapted.

In order for the SQLite3 support in PHP:
`sqlite3` has to be installed, as well as `php-sqlite` which also should pull in `pdo-sqlite` and or `pdo` for  usage.

*Note*: The prepared file is in the state of the latest database revision (currently 6) and can be used without any bootstrapping. Since we do currently not support the support for SQLite databases explicitly.
